### PR TITLE
[AIRFLOW-4997] Support for non-master branches

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4997

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

We need to build similar images for v1-10-stable and v1-10-tests and possibly for some future branches. We need to make sure that when v1-10-test PR/build is run in Travis, it uses relevant image (apache/airflow:v1-10-test-python...) and that when it is built on Dockerhub it pushes the right image.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Tested automatically by running all tests. 
### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: hooks/build [hunk 1]
```diff
@@ -155,10 +156,22 @@ export DOCKERHUB_REPO=${DOCKERHUB_REPO:="airflow"}
 export PYTHON_VERSION=\
 ${PYTHON_VERSION:=$(python -c 'import sys;print("%s.%s" % (sys.version_info.major, sys.version_info.minor))')}
 
+. ${MY_DIR}/_default_branch.sh
+
+# Source brnnch will be set in DockerHub
+SOURCE_BRANCH=${SOURCE_BRANCH:=${DEFAULT_BRANCH}}
+# if AIRFLOW_CONTAINER_BRANCH_NAME is not set it will be set to either SOURCE_BRANCH (if set in DockerHub
+# or default branch configured in _default_branch.sh
+AIRFLOW_CONTAINER_BRANCH_NAME=${AIRFLOW_CONTAINER_BRANCH_NAME:=${SOURCE_BRANCH}}
```
Comment (mik-laj) [inline hooks/build:165]: I found similarly looking code in the `scripts/ci//local_ci_stop_environment.sh` file. Do it not need a change? Probably not, but I prefer to ask.

Comment (potiuk) [reply hooks/build:165]: Not really. Stable branch is usually only built on Travis CI not for local developer builds. 

We usually work in the way that we make v1-10-test branch work locally  (where DEFAULT_BRANCH=v1-10-test is fine). And then we move stable branch up when v1-10-test branch is green. Then in effect stable branch is only run on Travis.

Comment (potiuk) [reply hooks/build:165]: Besides this is "stop" which is not really using the branch.

**Comment by milton0825** [comment]
I cherry picked this PR and I can verify that builds on my personal travis is now passing.

**Comment by potiuk** [comment]
I need to merge this one to fix a problem that forked Travis CI are broken.

**Pull request merged**